comp_xls.py

```diff
@@ -1,2 +1,48 @@
+import pandas as pd
+df1_e = pd.read_excel('699.xls')
+df1_e.to_csv('csv699.csv', index=False, sep=',', encoding='cp1251')
+df1 = pd.read_csv('csv699.csv', header=11, usecols=['Товар', 'Кількість'], encoding='cp1251', )
+df2 = pd.read_excel('export_products_230725.xlsx', header=1, sheet_name=0, usecols=['Товар', 'К-сть'])
+df2.rename(columns={'К-сть': "Кількість"}, inplace=True)
 
+for i in range(len(df2)):
+    df2['Кількість'][i] = df2['Кількість'][i].split()[0]
+
+for i in range(len(df1)):
+    df1['Кількість'][i] = str(df1['Кількість'][i]).split()[0].replace('.', ',')
+
+my_df = pd.DataFrame(columns=("Товар", "Кількість постачальника", "Кількість замовника"))
+
+for i in range(len(df2)):
+    name_article = df2['Товар'][i]
+    if name_article not in df1['Товар'].values:
+        my_df.loc[len(my_df)] = {"Товар": name_article, "Кількість постачальника": '0', "Кількість замовника": df2['Кількість'][i]}
+    else:
+        index_article = df1[df1["Товар"]==name_article].index[0]
+        if df2['Кількість'][i] != df1['Кількість'][index_article]:
+            my_df.loc[len(my_df)] = {"Товар": name_article, "Кількість постачальника": df1['Кількість'][index_article],
+                                     "Кількість замовника": df2['Кількість'][i]}
```

Comment:
try to avoid loops and comparison statements.
If you look at both files you will notice that both have table format. So just use rows and columns subset (for example df.iloc[8:-1, 3:26])